src/aggregator/aggregator/generic_elem.go

```diff
@@ -737,11 +745,9 @@ func (e *GenericElem) processValue(
 							l.Error("previous start time not in state map",
 								zap.Time("ts", ts))
 						})
-					} else if prevFlushState.consumedValues != nil {
-						// prev consumedValues may be null if the result was NaN.
-						prev.Value = prevFlushState.consumedValues[aggTypeIdx]
-						prev.TimeNanos = int64(prevFlushState.timestamp)
 					}
+					prev.Value = prevFlushState.consumedValues[aggTypeIdx]
```

Comment:
Why's this unnecessary now? I guess anything "previous" must have consumedVals? That'd only be the case if the previous had a binary op though right?

Reply:
see the comment below. 

the only time the consumed value of the previous aggregation could be nil, is if the consumed value was `NaN`. this can happen for certain aggregations like `Max` when there is no data yet. I removed the guard below that did  not populated `consumedValues` if the value was `NaN`, so this can never happen now. The code could already handle a `NaN` value, so it just seemed overly complicated. Removed 2 `if` statements that provided no value.

"That'd only be the case if the previous had a binary op though right?" - You're in the `binaryOp` case so you're guaranteed the previous was a binary op as well. The pipeline can't change within an aggregation (it would create a new aggregation).